lib/iris/cube.py

```diff
@@ -912,6 +928,37 @@ def add_aux_factory(self, aux_factory):
                             'iris.aux_factory.AuxCoordFactory.')
         self._aux_factories.append(aux_factory)
 
+    def add_cell_measure(self, cell_measure, data_dims=None):
+        """
+        Adds a CF cell measure to the cube.
+
+        Args:
+
+        * cell_measure
+            The :class:`iris.coords.CellMeasure`
+            instance to add to the cube.
+
+        Kwargs:
+
+        * data_dims
+            Integer or iterable of integers giving the data dimensions spanned
+            by the coordinate.
+
+        Raises a ValueError if a cell_measure with identical metadata already
+        exists on the cube.
+
+        See also
+        :meth:`Cube.remove_cell_measure()<iris.cube.Cube.remove_cell_measure>`.
+
+        """
+        if self.cell_measures(cell_measure):
+            raise ValueError('Duplicate cell_measures are not permitted.')
+        data_dims = self._check_multi_dim_metadata(cell_measure, data_dims,
+                                                   'cell_measure')
+        self._cell_measures_and_dims.append([cell_measure, data_dims])
+        self._cell_measures_and_dims.sort(key=lambda cmdims:
+                                          (cmdims[0]._as_defn(), cmdims[1]))
```

Comment:
@marqh I like @rhattersley style `key=lambda c_d: (c_d[0]._as_defn(), c_d[1]))` ... hints to the reader with regards to unpacking that `c_d` has two elements.